### ISSUE TITLE
update-kube-proxy from 1.17 to 1.18 Image not found error

### ISSUE THREAD

**Issue by composer22** [weaveworks/eksctl#4084]
Using 
```
eksctl version
0.60.0

eksctl utils update-kube-proxy --cluster=k8s-cluster --approve
```

Error - infinite loop in the pod
```
state:
  waiting:
    message: Back-off pulling image "602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/kube-proxy:v1.18.20-eksbuild.2"
    reason: ImagePullBackOff
```

Note: v1.18.20-eksbuild.2

which apparently doesn't exist

Documentation on AWS suggests image id:

https://docs.aws.amazon.com/eks/latest/userguide/managing-kube-proxy.html#kube-proxy-default-versions-table

```
1.18.8-eksbuild.1
```

This doc also outlines the newer approach 
```
eksctl update addon...
```

So, we have a conundrum of questions because docs might be out of date vs code not being correct, or both. And also what should the process be to upgrade from 1.17 old school to new, if at all. Can we use old school still here?

To sum up

Q: How to I patch around this in the daemon set? Which version should I use? 1.18.8-eksbuild.1, v1.18.20-eksbuild**.1**, some other flavour?

Q: Where did v1.18**.20-eksbuild.2** originate from in eksctl? It's clearly wrong.

Q: How can eksctl utils update-kube-proxy command validate this so it doesn't happen in the future, including giving adequate instructions to use a different approach from 1.17 to 1.18 and beyond?

Thanks

**Comment by composer22** [comment]
addendum question


https://github.com/weaveworks/eksctl/blob/4813c592e325e210d93e96d012f87ad9f66386ea/pkg/addons/default/kube_proxy.go#L164-L174

```
func kubeProxyImageTag(controlPlaneVersion string) (string, error) {
	isMinVersion, err := utils.IsMinVersion(api.Version1_18, controlPlaneVersion)
	if err != nil {
		return "", err
	}

	if isMinVersion {
		return fmt.Sprintf("v%s-eksbuild.2", controlPlaneVersion), nil
	}
	return fmt.Sprintf("v%s-eksbuild.1", controlPlaneVersion), nil
}

```

Clearly, 
```
isMinVersion, err := utils.IsMinVersion(api.Version1_18, controlPlaneVersion) 
```
and...

```
func IsMinVersion(minimumVersion, version string) (bool, error) {
	...
	return targetVersion.GE(minVersion), nil
}

```
e.g. return targetVersion.GE(minVersion), nil
is wrong since the table suggests eksbuild.1 for 1.18

**Comment by aclevername** [comment]
Thanks for opening the issue @composer22 . I've opened a PR to fix the incorrect build https://github.com/weaveworks/eksctl/pull/4086. I'm investigating why the version is wrong.

**Comment by aclevername** [comment]
Hi @composer22 , I believe the version is correct, it was just the build version causing the problem. For example `kube-proxy:v1.18.20-eksbuild.1` on eks 1.18 runs successful-:
```
  Normal  Scheduled  63s   default-scheduler  Successfully assigned kube-system/kube-proxy-2mwht to ip-192-168-28-178.us-west-2.compute.internal
  Normal  Pulling    63s   kubelet            Pulling image "<>.dkr.ecr.us-west-2.amazonaws.com/eks/kube-proxy:v1.18.20-eksbuild.1"
  Normal  Pulled     62s   kubelet            Successfully pulled image "<>.dkr.ecr.us-west-2.amazonaws.com/eks/kube-proxy:v1.18.20-eksbuild.1"
  Normal  Created    62s   kubelet            Created container kube-proxy
  Normal  Started    62s   kubelet            Started container kube-proxy
  ```
  
  closing this issue as https://github.com/weaveworks/eksctl/pull/4086 will go out in todays release

**Comment by composer22** [comment]
@aclevername Thanks for resolving this and for this feedback. I'll hopefully be able to patch this in and perform a recovery.

**Comment by composer22** [comment]
note as follow up

editing
```
kubectl edit daemonset kube-proxy -n kube-system
```

and changing image:

602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/kube-proxy:v1.18.20-eksbuild.2

to 

602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/kube-proxy:v1.18.20-eksbuild.1

worked.
